core/src/main/scala/kafka/coordinator/GroupMetadataManager.scala

```diff
@@ -150,7 +151,7 @@ class GroupMetadataManager(val brokerId: Int,
       // if we crash or leaders move) since the new leaders will still expire the consumers with heartbeat and
       // retry removing this group.
       val groupPartition = partitionFor(group.groupId)
-      val (magicValue, timestamp) = getMessageFormatVersionAndTimestamp(groupPartition)
+      val (magicValue, timestamp) = getMessageFormatVersionAndTimestampIfReplicaIsLocal(groupPartition)
```

Comment:
I think since we are getting rid of the group from the groupMetadataCache in the stopReplicaRequest on the old coordinator, we should not be hitting this ideally. But there looks like a race condition where in the stopReplica removes it from the cache after the evictGroupAndDeleteMetadata() has started running for that group.